src/odemis/driver/zeiss.py

```diff
@@ -985,13 +1071,38 @@ def _run_move(self):
                                      1)
                 traveled_dist = total_dist * traveled_ratio
                 self.pos = orig_pos + traveled_dist
+
+                # R is allowed to do any rotation, but always stays within 0 -> 360°
+                self.pos[-2] %= 2 * math.pi
+
             if not self._stage_stop.is_set():
                 self.pos = self.target_pos
+                self.pos[-2] %= 2 * math.pi
```

Comment:
pos is in degrees. So it should be `%= 360`.

Reply:
Actually the elements for the rotational axes in pos are converted to radians just before firing _run_move in _move_to_pos. This is to keep the readonly self._speed constant usable for the rotational axes. Therefore I have to check for 2π instead of 360°.

Reply:
Ah, thanks for explaining. I now see the conversions. So the advantage is that the realistic speed corresponding to mm/s also corresponds to a realistic speed in rad/s. 
With this in mind, I'd recommend to adjust the simulator to use degrees only. I think that the amount of code will stay about the same, but it'll be clearer that the rotations are always in degrees (in the simulator) and that the speed for rotation and linear movements is not the same. 
I'd suggest to have two speeds: `_speed_lin = 0.1 # mm/s ` and `_speed_rot = 10 # deg/s`. Then, you never need to do any conversion for the rotation axes... but you need to compute the move time slightly differently. It should be relatively straightforward. I'd suggest something like that:
```python
dur_lin = sum(numpy.sqrt((self.pos[[0,1,2,5]] - self.target_pos[[0,1,2,5]]) ** 2)) / self._speed_lin
dur_rot = sum(abs(self.pos[[3,4]] - self.target_pos[[3,4]])) / self._speed_rot
```